### PR TITLE
All changes done for my thesis

### LINKED ISSUE #9: Add more models
Check if the following models could be used for our setting. If so, add them.
I'll update the list as my research goes on.

 - [x] HAT-L (see https://arxiv.org/pdf/2205.04437v3.pdf)
 - [ ] CAR (see https://arxiv.org/pdf/1907.12904.pdf)
 - [x] SwinFIR (see https://arxiv.org/pdf/2208.11247v2.pdf)
 - [ ] RepSR (see https://arxiv.org/pdf/2205.05671.pdf)

### PR DESCRIPTION
**Pull request by bojobo**
- Added Dockerfile
- Switched from YAML to TOML for configuration files
  - Removed unused YAML files
  - Merged configuration of different models into one TOML file
  - Added config.py in which I use pydantic for the validation of the configuration files
- There is now one file handling all of the various datasets and datamodules
- The amount of returned data when loading a FITS file has been reduced
- The image from a FITS file is directly loaded into a torch.Tensor and subsequent modifications are adjusted
- The creation of the loss function is now more compact and easier to understand
- The VIFLoss has been implemented by torchmetrics, therefore I have switched to it
- The calculation of the metrics during training/validation/test has been adjusted to circumvent the need for an explicitly given clamp_max (this is the reason, why the ImageLogger is broken)
- Added new models
  - For SR: DRCT, SwinFir, SwinIr, HAT
  - For DN: Restormer
  - Since all these are transformers, I have extracted the common modules into transformer.modules and common tools into transformer.tools
- Moved the initialisation process of the models, so that it works as intended when using strategy DDP or FSDP
- After a successful fit run, the model will be automatically tested with the "best" checkpoint
- Reworked how the normalisation process works:
  - If clamp_max == 0, the image is scaled to [0, 1] based on its own max value
  - Otherwise clamp_max is used

The following is currently broken:
- ImageLogger
- XmmDisplayDataset 
- run_inference_on_file
- And probably xmmsas_tools

I will fix these issues with individual PR's

### CODE REVIEW

**Review by SamSweere — APPROVED** [review]

**Review by SamSweere — APPROVED** [review]
Massive merge, thanks for the continued work on this project. Hope it will be usefull for others in the future as well!
